### PR TITLE
[Doc] Add vSphere Ray cluster launcher user guide

### PR DESCRIPTION
**Pull request by wfangchi**
Similar as other providers, this change adds a user guide for vSphere Ray cluster launcher,
including how to prepare the vSphere environment and the frozen VM, as well as the general
steps to launch the cluster. It also contains a section on how to use vSAN File Service to
provision NFS endpoints as persistent storage for Ray AIR, with a new example YAML file.

In addition to that, existing examples and docs are updated to include the correct command
to install vSphere Python SDK.

Signed-off-by: Fangchi Wang <wfangchi@vmware.com>

## Why are these changes needed?

As mentioned in PR #39379 , we need a dedicated user guide for launching Ray clusters on vSphere. This change does that with a newly added vsphere.md, including a solution for Ray 2.7's deprecation of syncing to head node for Ray AIR, using VMware vSAN File Service.
## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+This guide details the steps needed to launch a Ray cluster in a vSphere environment.
+
+To start a vSphere Ray cluster, you will use the Ray cluster launcher with the VMware vSphere Automation SDK for Python.
+
+## Prepare the vSphere environment
+
+If you don't already have a vSphere deployment, you can learn more about it by reading the [vSphere documentation](https://docs.vmware.com/en/VMware-vSphere/index.html). The following prerequisites are needed in order to create Ray clusters.
+* [A vSphere cluster](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vcenter-esxi-management/GUID-F7818000-26E3-4E2A-93D2-FCDCE7114508.html) and [resource pools](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-resource-management/GUID-60077B40-66FF-4625-934A-641703ED7601.html) to host VMs composing Ray Clusters.
+* A network port group (either for a standard switch or distributed switch) or an NSX segment. VMs connected to this network should be able to obtain IP address via DHCP.
+* A datastore that can be accessed by all the host in the vSphere cluster.
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md:12]:
```suggestion
* A datastore that can be accessed by all the hosts in the vSphere cluster.
```

File: doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+This guide details the steps needed to launch a Ray cluster in a vSphere environment.
+
+To start a vSphere Ray cluster, you will use the Ray cluster launcher with the VMware vSphere Automation SDK for Python.
+
+## Prepare the vSphere environment
+
+If you don't already have a vSphere deployment, you can learn more about it by reading the [vSphere documentation](https://docs.vmware.com/en/VMware-vSphere/index.html). The following prerequisites are needed in order to create Ray clusters.
+* [A vSphere cluster](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vcenter-esxi-management/GUID-F7818000-26E3-4E2A-93D2-FCDCE7114508.html) and [resource pools](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-resource-management/GUID-60077B40-66FF-4625-934A-641703ED7601.html) to host VMs composing Ray Clusters.
+* A network port group (either for a standard switch or distributed switch) or an NSX segment. VMs connected to this network should be able to obtain IP address via DHCP.
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md:11]: Are there pages we can link to for more about "network port group", "switch" and "NSX segment"?

File: doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+## Prepare the frozen VM
+
+The vSphere Ray cluster launcher requires the vSphere environment to have a VM in a frozen state prior to deploying a Ray cluster. This VM is later used to rapidly create head and worker nodes with VMware's [instant clone](https://docs.vmware.com/en/VMware-vSphere/8.0/vsphere-vm-administration/GUID-853B1E2B-76CE-4240-A654-3806912820EB.html) technology. The internal details of the Ray cluster provisioning process using frozen VM can be found in this [Ray on vSphere architecture document](https://github.com/ray-project/ray/blob/master/python/ray/autoscaler/_private/vsphere/ARCHITECTURE.md). 
+
+You can follow [this document](https://via.vmw.com/frozen-vm-ovf) to create and set up the frozen VM. By default, Ray clusters' head and worker node VMs will be placed in the same resource pool as the frozen VM. When building and deploying the frozen VM, there are a couple of things to note:
+
+* The VM's network adapter should be connected to the port group or NSX segment configured in above section. And the `Connect At Power On` check box should be selected.
```
Comment (architkulkarni) [inline doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md:22]:
```suggestion
* The VM's network adapter should be connected to the port group or NSX segment configured in the above section. And the `Connect At Power On` check box should be selected.
```

File: python/ray/autoscaler/vsphere/defaults.yaml [hunk 4]
```diff
@@ -52,8 +52,9 @@ auth:
 # The node config specifies the launch config and physical instance type.
 available_node_types:
     ray.head.default:
-        # For example, {"CPU": 4, "Memory": 8192}
-        resources: {}
+        # The node type's CPU and GPU resources are by default the same as the frozen VM.
+        # You can override the resources here.
+        resources: {"CPU": 2}
```
Comment (architkulkarni) [inline python/ray/autoscaler/vsphere/defaults.yaml:57]: I think when the user uses a cluster launcher `YAML` file and omits a field, Ray will automatically fill in the value from this `defaults.yaml`.  Do we really want to automatically fill in `"CPU"": 2` if the user doesn't specify `resources`?

**Review by architkulkarni — COMMENTED** [review]
Looks good overall! Just a few minor questions

File: doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md (hunk 2 above)
Comment (wfangchi) [reply doc/source/cluster/vms/user-guides/launching-clusters/vsphere.md]: Yes. Added reference documentation pages for these elements.

File: python/ray/autoscaler/vsphere/defaults.yaml (hunk 4 above)
Comment (wfangchi) [reply python/ray/autoscaler/vsphere/defaults.yaml:57]: Yes. To keep the frozen VM's footprint small, the frozen VM is configured with only 1 CPU. So if not specified, the created head and worker node VMs will inherit this setting to be 1 CPU for each.

I checked the defaults.yaml for other providers. Looks like aws uses m5.large which is 2 vCPU. And azure and gcp both explicitly specify 2 CPUs. So we also put `"CPU"": 2` for ours.

**Review by architkulkarni — APPROVED** [review]

**Comment by wfangchi** [comment]
Hi @architkulkarni , kindly let us know if there is anything else we should add or amend. We'd like to include this guide in the 2.7 release 😄

**Comment by architkulkarni** [comment]
Failed tests: 
- learning_tests_stateless_cartpole_r2d2 unrelated
- rllib:examples/nested_action_spaces_ppo_torch unrelated
- Linkcheck error https://raw.githubusercontent.com/ray-project/ray/master/python/ray/autoscaler/vsphere/example-vsan-file-service.yaml will be fixed when this PR is merged
- Serve release tests and HA tests unrelated
- kuberay/test_autoscaling_e2e unrelated
- test_redis_tls unrelated

**Comment by architkulkarni** [comment]
> Hi @architkulkarni , kindly let us know if there is anything else we should add or amend. We'd like to include this guide in the 2.7 release 😄

Unfortunately the cherry-picks for 2.7 ended last Friday. However, you can always redirect users to the master branch docs https://docs.ray.io/en/master/cluster/vms/user-guides/launching-clusters/index.html#launching-vm-clusters (should be updated automatically in a few hours or a day)

**Pull request merged**